src/ts/utils/dom.ts

```diff
@@ -24,4 +40,66 @@ export function getInputEvent() {
         bubbles: true,
         cancelable: true,
     });
+}
+
+function addDelayAfterInput(fn: void, ms: number) {
+    return new Promise((res,rej) => {
+        fn;
+        setTimeout(res,ms);
+    })
+}
+
+function simulateUserInput(input: void) {
+    return addDelayAfterInput(input,20);
```

Comment:
just realized this (new to async things mysel) - 0 would be fine here, as we're in single-theaded env, so what we really want is for event loop "do other things" before coming back to us.
also correspondingly this should be part of key/mouse simulations themselves (will update my `keyboard` obj correspondingly)

Reply:
hmm, though, so 0 works fine for 1 press, but it's weird if you do multiple in short succession

Reply:
or it seems more related to whether I keep one of keys in combination pressed? weird 🤔

Reply:
I agree in principle with "0 ms" delay, but from my tests anything lower than 20 and things wouldn't execute correctly... of course it's possible some actions would take more than that really...

Reply:
Following your line of thinking on the event loop, I think long term solution is listening to dom changes that confirm our actions? Not sure how to do that atm

Reply:
I'm surprised about this one, need to dig deeper/understand this one better.   
I don't remember the exact name, but you can subscribe to Dom change events, so I imagine we can subscribe, waiting on the specific changes we want - i.e. "blue highlight" class appearing on the element or something like that

Reply:
MutationObserver, but it sucked when I tried that in Dynalist, was making the page laggy